Dockerfile

```diff
@@ -1,18 +1,82 @@
-FROM continuumio/miniconda3
+FROM nvidia/cuda:11.2.0-cudnn8-devel-ubuntu20.04 AS build
 
-WORKDIR /app
+# RNCAN certificate; uncomment (with right .cer name) if you are building behind a FW
+# COPY NRCan-RootCA.cer /usr/local/share/ca-certificates/cert.crt
+# RUN chmod 644 /usr/local/share/ca-certificates/cert.crt && update-ca-certificates
 
-# Create the environment:
-COPY environment.yml .
-RUN conda env create -f environment.yml
+ARG PYTHON_VERSION=3.9
+ARG CONDA_PYTHON_VERSION=3
+ARG CONDA_DIR=/opt/conda
 
-# Make RUN commands use the new environment:
-SHELL ["conda", "run", "-n", "geo_deep_env", "/bin/bash", "-c"]
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends git wget unzip bzip2 build-essential sudo \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-# Make sure the environment is activated:
-RUN echo "Make sure flask is installed:"
-RUN python -c "import flask"
+# Install miniconda
+ENV PATH $CONDA_DIR/bin:$PATH
+RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda$CONDA_PYTHON_VERSION-latest-Linux-x86_64.sh -O /tmp/miniconda.sh && \
+    echo 'export PATH=$CONDA_DIR/bin:$PATH' > /etc/profile.d/conda.sh && \
+    /bin/bash /tmp/miniconda.sh -b -p $CONDA_DIR && \
+    rm -rf /tmp/* && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# The code to run when container is started:
-#COPY run.py .
-ENTRYPOINT ["conda", "run", "--no-capture-output", "-n", "geo_deep_env", "python"]
\ No newline at end of file
+## Variables 
+ARG GIT_HASH=f7e6033
+
+RUN git clone "https://github.com/NRCan/geo-deep-learning.git" && \
+    cd geo-deep-learning && \
+    git checkout ${GIT_HASH}
+
+# TODO : create an environment for inference
+RUN conda env create -f geo-deep-learning/environment.yml && \
+    rm -rf .git data docs config example LICENSE README.md Dockerfile *.yml
```

Comment:
- the "config" folder is necessary and should not be removed in my opinion. 
- "example" folder contains an empty **22 bytes** README to be populated with an example usage in a notebook
- the "data" folder is necessary for a user to test GDL with the Massachusetts dataset locally. 

I don't think this rm -rf command is relevant other than for deleting .git history, which doesn't need to be important in the first place if we use git tags.

Reply:
If we use tags then what happens with CI if we want to dynamically build the imeage ?

Reply:
See [previous comment](https://github.com/NRCan/geo-deep-learning/pull/247#discussion_r797670749). CI will only break if dependencies change. When dependencies change, a new version should be tagged as good practice. Therefore the Dockerfile should be changed along with the tag.